### PR TITLE
Mind Batterers

### PR DESCRIPTION
**Pull request by ShadowDeath6**
Got this from @SeikoTheSeal
### Intent of your Pull Request

Adds mind batterers to the traitor uplink, for 7 TC. They have one use, but can be recharged manually by using it in-hand when it's out of charges. It takes 10 seconds to recharge.
#### Changelog

:cl: ShadowDeath6
rscadd: Adds Mind Batterers to traitor uplinks. It has a 50% chance per person to stun in an area around the user.
tweak: Mind batterers have one charge, but can be used in-hand when out of charge to recharge it. Recharge time takes 10 seconds.
/:cl:

I'm gonna be leaving for like 8 days so leave comments on how I can fix this or something.

### CODE REVIEW

**Comment by X-TheDark** [comment]
**Code stuff**
This isn't an issue with your code, but https://github.com/ShadowDeath6/yogstation/blob/aaaa/code/game/objects/items/devices/traitordevices.dm#L56 is basically useless. As, for all intents and purposes, Weaken = Stun (both are the same thing, but Weaken knocks you down), so giving a Stun with LESS duration than the Weaken is completely useless. Mind removing that if?

**Comments on the weapon**
**Absolutely against this** for the following reasons:
- RNG AoE stun that is impossible to protect against, since it just plain stuns you, no fortitude save;
- Range is 10, which is 3 more than your view range (which is 7) - you can be stunned by something that's outside your screen;
- Ignores view - which means you can get stunned from maintenance tunnels;
- You are dropped for between 20-40 seconds (Each tick of weaken is about 2 seconds);

I am not sure how to make this work, honestly. Especially since this is recharging. If this was 100% chance to work (but no recharging), you might see this as a last resort, but then, you basically have a guaranteed, uncounterable stun for 20-40 seconds. With recharging and RNG...I am just not sure how in the hell this will be balanced.

**Comment by ShadowDeath6** [comment]
1. I didn't do anything with the stuns, but I'll remove the stun prob since it's useless
2. I'll set the range to 5 or something
3. I like the idea of stunning through walls, it can lead to combos with thermals or something. 
4. BYOND ticks will always be weird, it shouldn't be as drastic as 2 seconds per tick, as tasers would weaken for like 10 seconds if that were true. 

You certainly can't spam it since it's one use before you wait 10 seconds to recharge it, and with a 50% chance to do anything I don't think it's wildly overpowered.

**Comment by ShadowDeath6** [comment]
Uh oh

**Comment by ShadowDeath6** [comment]
I'll fix it when I get home, online editor has failed me
Comment on the weapon if you have any concerns now that it's been updated

**Comment by Super3222** [comment]
This is literally stupid, and I know it is because I thought of making something a long long time ago. An RNG stun like THIS is completely unreliable and no one's going to buy it unless they are really that desperate.

What makes it even worse is that it stuns for a random chance to stun at a length that's minimum equal to a stunprod and MAXIMUM IS STRONGER THAN A FLASHBANG.

I don't need to tell you how terrible that is.

And stunning through walls isn't a good idea at all. It's like the same problem you have with the tesla "aaah! random thing just shocked me! aaah! now i'm ded". 

With this, literally anything stupid can happen and you can get away with murdering absolutely anyone you encounter.

Via these situations.

Walk up to the Captain, activate the RNG tool, he gets stunned for 5-15 seconds (which is more than enough time for) and than you pull out your energy sword and slash him to death. GG no re, it's a stunbaton that requires absolutely no skill or accuracy to use.

Wait inside of maintenance, see a random passenger run down, activate RNG tool, he gets stunned fo a random time, run out at him and slash him up.

Do you know what the problem is? It's unpredictable bullshit that gets you stunned. Remember the great words of the original powergamer, "Once you're stun, you're done!".

The only way I can see this working is if you made it LITERALLY charge up, and present a message like "Greytide McShitler raises the knob on the mind batterer" and in a few seconds flat everyone gets cucked. Something like that, something that can be predictable, something that won't get you completely cucked. "You feel a brief wave of pressure pass you as your brain becomes extremely agitated!".

And 50% chance for a stun that raises 5-15 (depending on how far you are from the blast), but if that fails than it does it for a small dosage at the maximum length (being like 5?).

**Comment by Super3222** [comment]
Or if the RNG chance for an RNG stun is still too powerful, than instead of being RNG, it should divide up the amount stun inside of it depending on how much people are around. Unless there's only one person, than it does it for a reduced time.

File: code/game/objects/items/devices/traitordevices.dm [hunk 1]
```diff
@@ -28,24 +28,30 @@ effective or pretty fucking useless.
...
 
-	for(var/mob/living/carbon/human/M in orange(10, user))
+	for(var/mob/living/carbon/human/M in orange(5, user))
 		spawn()
 			if(prob(50))
-
-				M.Weaken(rand(10,20))
-				if(prob(25))
-					M.Stun(rand(5,10))
+				M.Weaken(rand(5,15)
```
Comment (X-TheDark) [inline code/game/objects/items/devices/traitordevices.dm]: Missing right parenthesis, should be `M.Weaken(rand(5,15))` , that's what's causing Travis to fail with so many error messages.

**Comment by X-TheDark** [comment]
**TL;DR : I just don't like the concept of the weapon, really. Uncounterable stun that's either shit because of RNG or wins you the game because of RNG. The sad part is, I cannot think of any suggestions on how to improve this other than try it out after (hopefully) some more feedback on the idea.**

It's not one time use anymore, @Super3222 , it can be recharged by clicking on it when it's in your hand and after 10 seconds have passed (you can move during that time, since it uses spawn() ) it recharges.

Here's the thing, I don't mind if we try it out, but I don't foresee this going well. Either it's an unreliable and shit RNG stun that doesn't work or it's an RNG stun that just won you that greentext by pure luck.

The weaken of 5-15 seems fair'ish, for the first iteration (obviously, this can be changed). Also, I'm not sure if range of 5 is a bit too harsh. I was complaining about the fact that it stunned beyond the visible range. But I guess, since this can stun through walls, 5 seems reasonable enough.

**Comment by ShadowDeath6** [comment]
It'd be just as effective as stunprod+ esword. I can add a visible message to using it, but that might make it much less desirable to use. 

As for RNG, that's kind of the balance of the item. You can get lucky with a powerful semi ranged stun or you can get unlucky with it, but an 100% lel stun isn't fun to fight against.  It isn't like you can't use 4click death guns that are silent already.

**Comment by Super3222** [comment]
) ^:<

I have a lot more to say about it, but I'm going to go with X on the fact we should see how well this does. Which probably isn't going to be very well at all, but a test drive is definitely necessary. If it's recharging for a 50% automatic

**Comment by ShadowDeath6** [comment]
> :^(

**Comment by FluffySurvivor** [comment]
Jesus, how hard is it to understand you CANNOT balance something by slapping it with RNG ? That's the nature of randomness, you can't expect something to be constant/balanced by putting randomness to it. You may as well give it rng price while you're at it.
Sorry to tell, but this is dumb as shit.

**Comment by Notamaniac** [comment]
RNG is a horrible mechanic to rely on, especially at 50% which basically removes any chance of using it tactically i.e. Wanna bumrush an entire room full of sec? PAY THEE UNTO THE 50% RNG GOD OH WAIT I FAILED SHIT NOW EVERYONE KNOWS I'M A TRAITOR OH GOD THERE'S A STUN BATON BEING SHOVED UP MY A-

What this should do is simply confuse/(whateverflagitisthatcausesrandommovement) and cause people in the radius to drop whatever they're holding, giving you a nice non lethal option to escape sticky situations where you're surrounded.  It would be even more valuable if it worked on borgs, but that would be making it slightly OP

Anyways I thought yogs was rebasing, so new features are on the backburner for now?

**Comment by Super3222** [comment]
We are working on the rebase now ( as well as porting /tg/ bugfixes) and we are under a feature freeze (which doesn't halt us from making PR's about bugfixes, but halts features like these).

Consider this feature frozen until the rebase is done.

**Comment by oisin100** [comment]
If your going to continue this project then recreate this PR on the new repository.

**Comment by Super3222** [comment]
Next time make an issue before you make the PR, please.

**Comment by Super3222** [comment]
^ actually thats a terrible idea. don't do that. just make the pr

**Comment by ShadowDeath6** [comment]
wh
